libpod/image.go

```diff
@@ -50,29 +55,116 @@ type CopyOptions struct {
 // return will include the image, a false return will exclude it.
 type ImageFilter func(*storage.Image) bool
 
+type imagePullStruct struct {
+	imageName   string
+	tag         string
+	registry    string
+	hasRegistry bool
+	transport   string
+}
+
+func (ips imagePullStruct) returnFQName() string {
+	return fmt.Sprintf("%s%s/%s:%s", ips.transport, ips.registry, ips.imageName, ips.tag)
+
+}
+
+func getRegistriesToTry(image string) ([]types.ImageReference, error) {
+	var registries []types.ImageReference
+	var imageError = fmt.Sprintf("Unable to parse '%s'\n", image)
+	imgRef, err := reference.Parse(image)
+	if err != nil {
+		return registries, errors.Errorf(imageError)
+	}
+
+	tagged, istagged := imgRef.(reference.NamedTagged)
+	tag := "latest"
+	if istagged {
+		tag = tagged.Tag()
+	}
+	hasDomain := true
+	registry := reference.Domain(imgRef.(reference.Named))
+	if registry == "" {
+		hasDomain = false
+	}
+	imageName := reference.Path(imgRef.(reference.Named))
+	pImage := imagePullStruct{
+		imageName,
+		tag,
+		registry,
+		hasDomain,
+		DefaultRegistry,
+	}
+	if pImage.hasRegistry {
+		// If input has a registry, we have to assume they included an image
+		// name but maybe not a tag
+		pullRef, err := alltransports.ParseImageName(pImage.returnFQName())
+		if err != nil {
+			return registries, errors.Errorf(imageError)
+		}
+		registries = append(registries, pullRef)
+	} else {
+		// No registry means we check the globals registries configuration file
+		// and assemble a list of candidate sources to try
+		registryConfigPath := ""
```

Comment:
I we have a list of registries in the libpod runtime config - we should be checking those second (after environment variable but before the system defaults) I think.

Reply:
can you point to that? unless you are parsing the system registries elsewhere, there should be NO known registries.  @rhatdan  do you know about this

Reply:
We allow manually specifying registries when setting up libpod, in case someone wants to manually override system registries. We probably won't end up exposing this in kpod, but it's a good thing to have if you want to just use libpod

https://github.com/kubernetes-incubator/cri-o/blob/master/libpod/runtime.go#L37-L38

Reply:
ok, given I have moved this out of the libpod section and we are not exposing in kpod ... no changes to this piece of code.